File: calendar1_R&A_Algorithm/HPN JAVA R&A CODE/HpnCalendarTools/src/hpn/mutualExclusion/ricartAgrawala/RequestQueue.java
package hpn.mutualExclusion.ricartAgrawala;

import java.util.PriorityQueue;

public class RequestQueue
{

	private PriorityQueue<RequestObject> queue;
	public RequestQueue()
	{
		this.queue= new PriorityQueue<RequestObject>(10, new RequestObjectComparator());
	}
	public boolean add(RequestObject obj)
	{
		return this.queue.add(obj);
	}
	public boolean contains(RequestObject obj)
	{
		return this.queue.contains(obj);
	}
	public RequestObject head()
	{
		return this.queue.peek();
	}
	public boolean isEmpty()
	{
		return this.queue.isEmpty();
	}
	public RequestObject remove()
	{
		return this.queue.poll();
	}	
}

File: calendar1_R&A_Algorithm/HPN C# R&A CODE/HpnCalendarTools/HpnCalendarTools/hpn/mutualExclusion/ricartAgrawala/RequestQueue.cs
using hpn.dataStructures.queue;
using System;

namespace hpn.mutualExclusion.ricartAgrawala
{
    public class RequestQueue
    {

        private PriorityQueue<RequestObject> queue;
        public RequestQueue()
        {
            this.queue = new PriorityQueue<RequestObject>();
        }
        public bool add(RequestObject obj)
        {
            return this.queue.Add(obj);
        }
        public bool contains(RequestObject obj)
        {
            if (this.isEmpty())
                return false;
            else
                return this.queue.Contains(obj);
        }
        public RequestObject head()
        {
            if (this.isEmpty())
                return null;
            else
                return this.queue.Peek();
        }
        public bool isEmpty()
        {
            return this.queue.IsEmpty;
        }
        public RequestObject remove()
        {
            if (this.isEmpty())
                return null;
            else
                return this.queue.Poll();                
        }
    }
}
